### PR TITLE
Contribution flow: use a collective picker in logged in step profile

### PR DESCRIPTION
**Pull request by Betree**
Part of https://github.com/opencollective/opencollective/issues/4692
Resolve https://github.com/opencollective/opencollective/issues/3628
Implements the design from https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/%5BOC-Design%5D-Production-Ready-%E2%9C%85?node-id=13838%3A161523

https://user-images.githubusercontent.com/1556356/146789206-36db85d5-c45b-40d4-a6be-88e10c8a10ae.mp4

### CODE REVIEW

File: components/contribution-flow/ContributeProfilePicker.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+const { USER, ORGANIZATION, COLLECTIVE, FUND, EVENT, PROJECT } = CollectiveType;
+
+const filterProfiles = (profiles, account, canUseIncognito) => {
+  return profiles.filter(p => {
+    // if admin of collective you are donating to, remove it from the list
+    if (p.id === account.legacyId) {
+      return false;
+    } else if (!canUseIncognito && p.isIncognito) {
+      return false;
+    } else if (p.type === 'COLLECTIVE' && (!p.host || p.host.id !== account.host?.legacyId)) {
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:21]:
```suggestion
    } else if (p.type === CollectiveType.COLLECTIVE && (!p.host || p.host.id !== account.host?.legacyId)) {
```

File: components/contribution-flow/ContributeProfilePicker.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  const getOptionFromAccount = value => ({ [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true, value, label: value.name });
+  const sortOptions = options => sortBy(options, 'value.name');
+  const profileOptions = profiles.map(getOptionFromAccount);
+  const profilesByType = groupBy(profileOptions, p => p.value.type);
+  const myself = profilesByType[USER] || [];
+  const myOrganizations = sortOptions(profilesByType[ORGANIZATION] || []);
+
+  // Add incognito profile entry if it doesn't exists
+  if (canUseIncognito) {
+    const hasIncognitoProfile = profiles.some(p => p.type === 'USER' && p.isIncognito);
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:45]:
```suggestion
    const hasIncognitoProfile = profiles.some(p => p.type === CollectiveType.USER && p.isIncognito);
```

File: components/contribution-flow/ContributeProfilePicker.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    if (!hasIncognitoProfile) {
+      // TODO Test this with a new user
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:47]:
```suggestion
```

Tested with a new user and seems to work fine. 👍🏽

File: components/contribution-flow/ContributeProfilePicker.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      myself.push(
+        getOptionFromAccount({
+          id: 'incognito',
+          type: 'USER',
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:51]:
```suggestion
          type: CollectiveType.USER,
```

File: components/contribution-flow/ContributeProfilePicker.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+          isIncognito: true,
+          name: intl.formatMessage({ id: 'profile.incognito', defaultMessage: 'Incognito' }),
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:53]: Currently we show something like this; 

image

Note that it says, `Check privacy box for info`. This doesn't seem to make sense I think. If what's meant by privacy box is the blue box beneath the collective picker, it doesn't mention anything about incognito contributions. The FAQ on the right side does mention about incognito contributions, maybe we can change the line to say, "Check FAQ for more info"; wdyt? 🤔

File: components/contribution-flow/ContributeProfilePicker.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  return (
+    <CollectivePicker
+      data-cy="contribute-profile-picker"
+      collective={selectedProfile}
+      addLoggedInUserAsAdmin
+      options={options}
+      isSearchable={filteredProfiles.length > 8}
+      creatable
+      excludeAdminFields
+      types={['ORGANIZATION']}
```
Comment (SudharakaP) [inline components/contribution-flow/ContributeProfilePicker.js:145]:
```suggestion
      types={[CollectiveType.ORGANIZATION]}
```

File: components/contribution-flow/ContributionFlowStepContainer.js [hunk 7]
```diff
@@ -108,6 +108,7 @@ class ContributionFlowStepContainer extends React.Component {
   /** Return an array of any other associated profile the user might control */
   getOtherProfiles() {
     const { LoggedInUser, collective } = this.props;
+    console.log('Get profiles from', LoggedInUser);
```
Comment (SudharakaP) [inline components/contribution-flow/ContributionFlowStepContainer.js:111]:
```suggestion
```

**Review by SudharakaP — COMMENTED** [review]
Some minor things I noticed. I'll go through this again. 😉

**Review by SudharakaP — COMMENTED** [review]
One other thing is that in mobile the content of the collective picker is cutoff. It would be nice I think if we can make it wrap. 😉  

image

File: components/contribution-flow/ContributeProfilePicker.js (hunk 5 above)
Comment (Betree) [reply components/contribution-flow/ContributeProfilePicker.js:53]: Good catch, it doesn't make much sense ideed. I'm going to rephrase this whole thing

**Pull request merged**